drake/systems/System.h

```diff
@@ -48,14 +51,105 @@ namespace Drake {
  * @}
  */
 
-  template <typename System>
-  struct SystemSizeTraits {
-    const static int num_states = System::template StateVector<double>::RowsAtCompileTime;
-    const static int num_inputs = System::template InputVector<double>::RowsAtCompileTime;
-    const static int num_outputs = System::template OutputVector<double>::RowsAtCompileTime;
-    static_assert(num_states >= 0, "still need to handle the variable-size case");
-    static_assert(num_inputs >= 0, "still need to handle the variable-size case");
-    static_assert(num_outputs >= 0, "still need to handle the variable-size case");
+  namespace internal {
+    template<typename System, bool Enable = false>
+    struct NumStatesDispatch {
+      static std::size_t eval(const System &sys) { return System::template StateVector<double>::RowsAtCompileTime; }
+    };
+
+    template<typename System>
+    struct NumStatesDispatch<System,true > {
+      static std::size_t eval(const System &sys) { return sys.getNumStates(); }
+    };
```

Comment:
i did this because MSVC 2013 was failing to compile the two enable_if logic, saying "too complex or recursive template aliasing".  MSVC was fine with either.